client/http-client/src/transport.rs

```diff
@@ -98,35 +134,30 @@ impl HttpTransportClient {
 			}
 		};
 
-		// Cache request headers: 2 default headers, followed by user custom headers.
-		// Maintain order for headers in case of duplicate keys:
-		// https://datatracker.ietf.org/doc/html/rfc7230#section-3.2.2
-		let mut cached_headers = HeaderMap::with_capacity(2 + headers.len());
-		cached_headers.insert(hyper::header::CONTENT_TYPE, HeaderValue::from_static(CONTENT_TYPE_JSON));
-		cached_headers.insert(hyper::header::ACCEPT, HeaderValue::from_static(CONTENT_TYPE_JSON));
```

Comment:
Dumb question: Would we need a http::middleware in substrate to insert those two headers?

Reply:
I can move that back, I didn't like the `cached headers fn` but did it when I was experimenting with the conditional compilation

Reply:
Not sure but these can be overwritten by custom headers or middleware :)

Reply:
I like it how it is now 👍 I missed it during my first review pass and then it made sense for me in the second run 😄 thanks!